x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/create_rules_route.test.ts

```diff
@@ -130,5 +137,44 @@ describe('create_rules', () => {
       const { statusCode } = await server.inject(request);
       expect(statusCode).toBe(400);
     });
+
+    test('catches error if createRules throws error', async () => {
+      clients.actionsClient.create.mockResolvedValue(createActionResult());
+      clients.alertsClient.find.mockResolvedValue(getFindResult());
+      clients.alertsClient.get.mockResolvedValue(getResult());
+      clients.alertsClient.create.mockResolvedValue(getResult());
+      clients.savedObjectsClient.find.mockResolvedValue(getFindResultStatus());
```

Comment:
As some of these become repeating patterns of: 

```
      clients.actionsClient.create.mockResolvedValue(createActionResult());
      clients.alertsClient.find.mockResolvedValue(getFindResult());
      clients.alertsClient.get.mockResolvedValue(getResult());
      clients.alertsClient.create.mockResolvedValue(getResult());
      clients.savedObjectsClient.find.mockResolvedValue(getFindResultStatus());
```

I wonder if they could be moved into a test utils or moved into a local function for this file? Optional, just thinking of ways we could reduce boilerplate as we gain collections of tests. Sometimes you can't as each block of code is just so _slightly_ different than the other ones.

Reply:
@FrankHassanabad @dhurley14 since this is all changing in #58292 my preference would be to defer any significant refactoring here.

To your point, though, my plan is to define a `createMockRequestHandlerContext` helper that's prepopulated with the plugin-defined mocks. As tests need to override these behaviors, they can do so in a similar manner to what we have now:

```ts
describe('a test', () => {
  let context = createMockRequestHandlerContext();

  beforeEach(() => {
    context = createMockRequestHandlerContext();
    context.alerting.getAlertsClient.mockReturnValue(myMockAlertsClient);
  });

  it('works', () => {
    const request = httpServerMock.createKibanaRequest();
    const responseSpy = { ok: jest.fn() };
    handler(context, request, responseSpy);
    
    expect(responseSpy.ok).toHaveBeenCalledWith(...);
  });
});
```

While I can't yet speak to the extend of the issue, some cursory investigation showed that some of these mocked values are unnecessary (I commented out a few mocks in a few tests, and most of the tests continued to pass); my hope is that these third-party mocks will cut down significantly on the verbosity of these tests.

Reply:
Sounds good @rylnd , totally agree.